docs/api.md

```diff
@@ -43,7 +43,8 @@ ClusterSpec defines the desired state for a M3 cluster to be converge to.
 | numberOfShards | NumberOfShards defines how many shards in total | int32 | false |
 | isolationGroups | IsolationGroups specifies a map of key-value pairs. Defines which isolation groups to deploy persistent volumes for data nodes | [][IsolationGroup](#isolationgroup) | false |
 | namespaces | Namespaces specifies the namespaces this cluster will hold. | [][Namespace](#namespace) | false |
-| configMapName | ConfigMapName specifies the ConfigMap to use for this cluster. If unset a sane default will be used. | *string | false |
+| etcdEndpoints | EtcdEndpoints defines the etcd endpoints to use for service discovery. Must be set if no custom configmap is defined. If set, etcd endpoints will be templated in to the default configmap template. | []string | false |
+| configMapName | ConfigMapName specifies the ConfigMap to use for this cluster. If unset a default configmap with template variables for etcd endpoints will be used. | *string | false |
```

Comment:
should we link the default template that will be used as `default-config` above?

Reply:
Ah yeah good point. I think I'll add a comment saying "see configuration section of docs" just so we don't have too many links in the comments